sklearn/utils/_test_common/instance_generator.py

```diff
@@ -725,3 +736,375 @@ def _yield_instances_for_check(check, estimator_orig):
+        "check_fit2d_1sample": "_scale_normalize fails",
+        "check_fit2d_1feature": "raises apply_along_axis error",
+        "check_estimator_sparse_matrix": "does not fail gracefully",
+        "check_estimator_sparse_array": "does not fail gracefully",
+        "check_methods_subset_invariance": "empty array passed inside",
+        "check_dont_overwrite_parameters": "empty array passed inside",
+        "check_fit2d_predict1d": "empty array passed inside",
+        # ValueError: Found array with 0 feature(s) (shape=(23, 0))
+        # while a minimum of 1 is required.
+        "check_dict_unchanged": "FIXME",
+    },
+    SpectralBiclustering: {
+        "check_estimators_dtypes": "raises nan error",
+        "check_fit2d_1sample": "_scale_normalize fails",
+        "check_fit2d_1feature": "raises apply_along_axis error",
+        "check_estimator_sparse_matrix": "does not fail gracefully",
+        "check_estimator_sparse_array": "does not fail gracefully",
+        "check_methods_subset_invariance": "empty array passed inside",
+        "check_dont_overwrite_parameters": "empty array passed inside",
+        "check_fit2d_predict1d": "empty array passed inside",
+    },
+    SplineTransformer: {
+        "check_estimators_pickle": (
+            "Current Scipy implementation of _bsplines does not"
+            "support const memory views."
+        ),
+    },
+    SVC: {
+        # TODO: fix sample_weight handling of this estimator when probability=False
+        # TODO: replace by a statistical test when probability=True
+        # see meta-issue #16298
+        "check_sample_weight_equivalence": (
+            "sample_weight is not equivalent to removing/repeating samples."
+        ),
+    },
+    SVR: {
+        # TODO: fix sample_weight handling of this estimator, see meta-issue #16298
+        "check_sample_weight_equivalence": (
+            "sample_weight is not equivalent to removing/repeating samples."
+        ),
+    },
+    TunedThresholdClassifierCV: {
+        "check_classifiers_train": "Threshold at probability 0.5 does not hold",
+        "check_sample_weight_equivalence": (
+            "Due to the cross-validation and sample ordering, removing a sample"
+            " is not strictly equal to putting is weight to zero. Specific unit"
+            " tests are added for TunedThresholdClassifierCV specifically."
+        ),
+    },
+}
+
+
+def _get_expected_failed_checks(estimator):
+    """Get the expected failed checks for an estimator."""
+    failed_checks = PER_ESTIMATOR_XFAIL_CHECKS.get(type(estimator), {})
+
+    tags = get_tags(estimator)
+    if type(estimator) in [KNeighborsClassifier, KNeighborsRegressor]:
+        if tags.input_tags.pairwise:
+            failed_checks.update(
+                {
+                    "check_n_features_in_after_fitting": "FIXME",
+                    "check_dataframe_column_names_consistency": "FIXME",
+                }
+            )
```

Comment:
all xfail marks that depend on the instance, come here. As of now, we have only these two cases.

Reply:
It should be written as a comment since this is a bit more tricky case.